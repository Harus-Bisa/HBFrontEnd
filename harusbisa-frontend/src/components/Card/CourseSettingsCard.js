import React from "react";
import {connect} from "react-redux";
import { Button } from "reactstrap";

function CourseSettingsCard(props){
    return(
        <div style={{margin:'30px 0px'}}>
            <div className="row">
                <div className="col-8">
                    <h5>{props.course.courseName}</h5>
                    <p>{props.course.instructors[0]}</p>
                    <p>{props.course.startTerm} - {props.course.endTerm}</p>
                    <p>Kode Bergabung: {props.course.joinCode}</p>
                </div>
                <div className="col-4">
                    <Button className="neutral-button" style={{width:'100%'}}>Drop kelas</Button>
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state, ownProps){
    var id = ownProps.id

    for (let i=0; i<state.courses.length; i++){
        if (state.courses[i].courseId === id){
            return {
                course: state.courses[i]
            }
        }
    }
    return {}
}

export default connect(mapStateToProps)(CourseSettingsCard)

