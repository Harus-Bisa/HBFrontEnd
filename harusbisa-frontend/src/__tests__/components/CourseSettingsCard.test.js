import React from "react";
import {mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {Provider} from "react-redux";
import CourseSettingsCard from "../../components/Card/CourseSettingsCard";

const mockStore = configureMockStore();
describe("CourseSettingsCard component", () =>{
    let initialState;
    let component;
    let store;
    beforeEach(() =>{
        initialState = {
            courses:[{
                "courseId": "5db492c349a67b33b8d0a2a2",
                "courseName": "Calculus I",
                "courseDescription": "The basic of calculus (derivatives).",
                "startTerm": "August 2019",
                "endTerm": "December 2019",
                "numberOfLectures": 5,
                "numberOfStudents": 21,
                "instructors": [
                    "John Doe",
                    "Bryan Smith"
                ],
                "joinCode": "123654"
              }]
        }
        store = mockStore(initialState);
        component = mount(
        <Provider store={store}>
            <CourseSettingsCard id={initialState.courses[0].courseId}/>
        </Provider>)
    })

    it("Matches snapshot", () =>{
        expect(component.html()).toMatchSnapshot();
    })
})