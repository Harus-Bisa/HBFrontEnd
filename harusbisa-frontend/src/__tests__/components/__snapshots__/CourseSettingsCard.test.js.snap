// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSettingsCard component Matches snapshot 1`] = `"<div style=\\"margin: 30px 0px;\\"><div class=\\"row\\"><div class=\\"col-8\\"><h5>Calculus I</h5><p>John Doe</p><p>August 2019 - December 2019</p><p>Kode Bergabung: 123654</p></div><div class=\\"col-4\\"><button style=\\"width: 100%;\\" class=\\"neutral-button btn btn-secondary\\">Drop kelas</button></div></div></div>"`;
